### PR TITLE
Post branch

### PR DESCRIPTION
**Pull request by champl0o**

### CODE REVIEW

File: app/assets/stylesheets/posts.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Place all the styles related to the Posts controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
+
+ul {
+    .list-inline-item {
```
Comment (afurm) [inline app/assets/stylesheets/posts.scss:6]: can you please fix spaces here?

File: app/assets/stylesheets/scaffolds.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  border: 0;
+}
+
+div.actions {
+  margin-bottom: 10px;
+  display: inline;  
+}
+
+.return-inline {
+    display: inline;
```
Comment (afurm) [inline app/assets/stylesheets/scaffolds.scss:12]: same here

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Post < ApplicationRecord
+    has_and_belongs_to_many :tags
```
Comment (afurm) [inline app/models/post.rb:2]: spaces

File: app/models/tag.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class Tag < ApplicationRecord
+    has_and_belongs_to_many :posts
```
Comment (afurm) [inline app/models/tag.rb:2]: see above comment

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  def index
+    @posts = Post.all.order(:created_at)
+  end
+
+  def search
+    if params[:search].blank?
+      redirect_to root_path and return
+    else
+      @parameter = params[:search].downcase
+      @results = Post.all.where("lower(title) LIKE :search", search: "%#{@parameter}%")
```
Comment (afurm) [inline app/controllers/posts_controller.rb:14]: Can this `Post.all.where("lower(title) LIKE :search", search: "%#{@parameter}%")
` be rewritten like  Post.search_result(@parameter)?

File: app/views/posts/_header.html.erb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<nav class="navbar navbar-expand-lg py-4 navbar-light bg-light">
+    <div class="container-fluid">
```
Comment (afurm) [inline app/views/posts/_header.html.erb:2]: spaces

File: app/views/posts/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+          <% post.tags.each do |tag| %>
+              <li class="list-inline-item"><%= tag.title %> </li>
+          <% end %>
+        </ul>
+      </div>
+    </div>
+    <% end %>
+<br>
+
+
```
Comment (afurm) [inline app/views/posts/index.html.erb:26]: space

File: db/migrate/20210315131535_create_join_table_posts_tags.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+class CreateJoinTablePostsTags < ActiveRecord::Migration[6.1]
+  def change
+    create_join_table :posts, :tags do |t|
+      # t.index [:post_id, :tag_id]
+      t.index [:tag_id, :post_id]
```
Comment (afurm) [inline db/migrate/20210315131535_create_join_table_posts_tags.rb:5]: why one is comment and the other is not?

**Review by afurm — APPROVED** [review]
Just a few comments, otherwise looks good. Also, if you have an empty models/tests you can remove them.

File: app/models/post.rb (hunk 3 above)
Comment (champl0o) [reply app/models/post.rb:2]: what is the problem with spaces in this file?

Comment (afurm) [reply app/models/post.rb:2]: See the difference between two files: 
Screenshot 2021-03-16 at 11 05 36
Screenshot 2021-03-16 at 11 05 26

**Pull request merged**